packages/cli-kit/src/haiku.test.ts

```diff
@@ -0,0 +1,29 @@
+import {inTemporaryDirectory, write} from './file.js'
+import {join} from './path.js'
+import {generate} from './haiku.js'
+import {describe, expect, test} from 'vitest'
+
+describe('generate', () => {
+  test('rerolls the name if a directory exists with the same name', async () => {
+    // make randonmess predictable and pick from the array of adjectives and nouns
```

Comment:
Actually I discovered there is a better way to do this with function spies, I'll refactor it with that.